src/Test.UnitTests.Sarif/Readers/NonSeekableStream.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Readers
+{
+    internal class NonSeekableStream : Stream
```

Comment:
>NonSeekableStream [](http://example.com/codeflow?start=19&length=17)

Please delete this class, which seems strange. I'm not sure that it covers all cases properly. For example, most non-seekable streams don't allow resetting the length, which you do here. 

Please go find an actual non-seekable stream in .NET and use it. 

Try System.IO.Packaging, I think their compressed streams aren't seekable.

[https://docs.microsoft.com/en-us/dotnet/api/system.io.packaging.storageinfo.createstream?view=windowsdesktop-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.io.packaging.storageinfo.createstream?view=windowsdesktop-5.0)

Reply:
that package is only for `Windows Desktop 5`.

the length isn't settable. Is is only a get.

Will search for a non-seekable stream.

Reply:
Replaced  with a cryptoStream and removed the nonseekable class